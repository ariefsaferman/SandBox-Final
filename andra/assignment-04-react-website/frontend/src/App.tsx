import { Route, Routes } from "react-router-dom";

import MainLayout from "./components/MainLayout";
import Home from "./pages/Home";
import Login from "./pages/Login";
import Register from "./pages/Register";

import "./App.css";
import "bootstrap/dist/css/bootstrap.min.css";
import "react-toastify/dist/ReactToastify.css";
import "@fontsource/montserrat";
import "bootstrap/dist/js/bootstrap.min.js";
import Topup from "./pages/Topup";
import Transfer from "./pages/Transfer";
import ProtectedPage from "./pages/ProtectedPage";
import UnauthenticatedPage from "./pages/UnauthenticatedPage";
import Games from "./pages/Games";

function App() {
  return (
    <div className="App">
      <Routes>
        <Route element={<MainLayout />}>
          <Route element={<UnauthenticatedPage />}>
            <Route path="/login" element={<Login />} />
            <Route path="/register" element={<Register />} />
          </Route>
          <Route path="/" element={<ProtectedPage />}>
            <Route index element={<Home />} />
            <Route path="/topup" element={<Topup />} />
            <Route path="/transfer" element={<Transfer />} />
            <Route path="/games" element={<Games />} />
          </Route>
        </Route>
        <Route path="*" element={<h2>404 Not Found</h2>} />
      </Routes>
    </div>
  );
}

export default App;
