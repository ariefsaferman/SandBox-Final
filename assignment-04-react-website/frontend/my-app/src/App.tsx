import React from "react";
import LoginPage from "./pages/LoginPage/index";
import { Route, Routes } from "react-router-dom";
import "./App.css";
import "https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/js/bootstrap.bundle.min.js";
import "bootstrap/dist/css/bootstrap.min.css";
import RegisterPage from "./pages/RegisterPage";
import HomePage from "./pages/HomePage";
import ProtectedPage, { ProtectReversePage } from "./pages/ProtectedPage";
import TransferPage from "./pages/TransferPage";
import TopupPage from "./pages/TopupPage";
import GamesPage from "./pages/GamesPage";

import "react-toastify/dist/ReactToastify.css";

function App() {
  return (
    <div className="App">
      <Routes>
        <Route element={<ProtectedPage />}>
          <Route path="/" element={<HomePage />}></Route>
          <Route path="/transfer" element={<TransferPage />}></Route>
          <Route path="/topup" element={<TopupPage />}></Route>
          <Route path="/games" element={<GamesPage />}></Route>
        </Route>
        <Route element={<ProtectReversePage />}>
          <Route path="/login" element={<LoginPage />}></Route>
          <Route path="/register" element={<RegisterPage />}></Route>
        </Route>
        <Route path="*" element={<h2> not found page</h2>}></Route>
      </Routes>
    </div>
  );
}

export default App;
