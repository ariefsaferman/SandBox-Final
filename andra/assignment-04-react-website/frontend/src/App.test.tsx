import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "./App";

test("renders", () => {
  const view = render(<App />, { wrapper: BrowserRouter });
  expect(view).toMatchSnapshot();
});
