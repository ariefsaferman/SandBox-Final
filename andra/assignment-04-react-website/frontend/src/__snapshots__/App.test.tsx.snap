// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div>
          <nav
            class="navbar navbar-expand-lg bg-transparent fixed-top"
          >
            <div
              class="container"
            >
              <h1
                class="navbar-brand m-0"
              >
                DigiWallet
              </h1>
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                class="navbar-toggler"
                data-bs-target="#navbarSupportedContent"
                data-bs-toggle="collapse"
                type="button"
              >
                <span
                  class="navbar-toggler-icon"
                />
              </button>
              <div
                class="collapse navbar-collapse justify-content-end"
                id="navbarSupportedContent"
              >
                <ul
                  class="navbar-nav"
                >
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    class="nav-item ms-lg-5"
                  >
                    <a
                      aria-current="page"
                      class="nav-link text-white active"
                      href="/login"
                    >
                      Login
                    </a>
                  </li>
                  <li
                    class="nav-item ms-lg-5"
                  >
                    <a
                      class="nav-link text-white"
                      href="/register"
                    >
                      Register
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
          <div
            class="container"
          >
            <div
              class="row vh-100"
            >
              <div
                class="col-lg-6 d-flex flex-column justify-content-center"
              >
                <h2
                  class="title"
                >
                  Login
                </h2>
                <form
                  class="form__login"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="fw-bold"
                      for="email"
                    >
                      Email
                    </label>
                    <input
                      class="form-control form-control-lg"
                      id="email"
                      name="email"
                      placeholder="asep.bc@gmail.com"
                      type="email"
                    />
                  </div>
                  <div
                    class="form-group"
                  >
                    <label
                      class="fw-bold"
                      for="password"
                    >
                      Password
                    </label>
                    <input
                      class="form-control form-control-lg"
                      id="password"
                      name="password"
                      placeholder="********"
                      type="password"
                    />
                  </div>
                  <button
                    class="btn btn-lg w-100"
                    type="submit"
                  >
                    Submit
                  </button>
                </form>
              </div>
              <div
                class="col-lg-6 auth__col"
              >
                <div
                  class="auth__hero"
                >
                  <img
                    alt="Auth"
                    src="undraw_login_re_4vu2 1.png"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div>
        <nav
          class="navbar navbar-expand-lg bg-transparent fixed-top"
        >
          <div
            class="container"
          >
            <h1
              class="navbar-brand m-0"
            >
              DigiWallet
            </h1>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler"
              data-bs-target="#navbarSupportedContent"
              data-bs-toggle="collapse"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="collapse navbar-collapse justify-content-end"
              id="navbarSupportedContent"
            >
              <ul
                class="navbar-nav"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="nav-item ms-lg-5"
                >
                  <a
                    aria-current="page"
                    class="nav-link text-white active"
                    href="/login"
                  >
                    Login
                  </a>
                </li>
                <li
                  class="nav-item ms-lg-5"
                >
                  <a
                    class="nav-link text-white"
                    href="/register"
                  >
                    Register
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
        <div
          class="container"
        >
          <div
            class="row vh-100"
          >
            <div
              class="col-lg-6 d-flex flex-column justify-content-center"
            >
              <h2
                class="title"
              >
                Login
              </h2>
              <form
                class="form__login"
              >
                <div
                  class="form-group"
                >
                  <label
                    class="fw-bold"
                    for="email"
                  >
                    Email
                  </label>
                  <input
                    class="form-control form-control-lg"
                    id="email"
                    name="email"
                    placeholder="asep.bc@gmail.com"
                    type="email"
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="fw-bold"
                    for="password"
                  >
                    Password
                  </label>
                  <input
                    class="form-control form-control-lg"
                    id="password"
                    name="password"
                    placeholder="********"
                    type="password"
                  />
                </div>
                <button
                  class="btn btn-lg w-100"
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>
            <div
              class="col-lg-6 auth__col"
            >
              <div
                class="auth__hero"
              >
                <img
                  alt="Auth"
                  src="undraw_login_re_4vu2 1.png"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
